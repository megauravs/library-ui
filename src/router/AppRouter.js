import React, { useEffect, useState } from 'react';
import axios from 'axios';
import { BrowserRouter, Switch, Route, Redirect } from 'react-router-dom';
import Header from '../components/Header';
import AddBook from '../components/AddBook';
import BooksList from '../components/BooksList';
import useLocalStorage from '../hooks/useLocalStorage';
import EditBook from '../components/EditBook';
import BooksContext from '../context/BooksContext';

const AppRouter = () => {
  const [books, setBooks] = useState([]);
  const [errorMessage, setErrorMessage] = useState('');

  useEffect(() => {
    axios.get("http://localhost:8080/api/v1/books")
    .then(response => setBooks(response.data))
    .catch(error => {
      setErrorMessage(error);
      console.error('There was an error!', error.message);
  })}, []);

  return (
    <BrowserRouter>
      <div>
        <Header />
        <div className="main-content">
          <BooksContext.Provider value={{ books, setBooks }}>
            <Switch>
              <Route component={BooksList} path="/" exact={true} />
              <Route component={AddBook} path="/add" />
              <Route component={EditBook} path="/edit/:id" />
              <Route component={() => <Redirect to="/" />} />
            </Switch>
          </BooksContext.Provider>
        </div>
      </div>
    </BrowserRouter>
  );
};

export default AppRouter;