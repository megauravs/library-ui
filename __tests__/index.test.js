import AppRouter from '../src/router/AppRouter';
import React from 'react';
import { mount } from 'enzyme';

test('TodoComponent renders the text inside it', () => {
    const wrapper = mount(
      <AppRouter />
    );
    expect(wrapper).toMatchSnapshot();
  });