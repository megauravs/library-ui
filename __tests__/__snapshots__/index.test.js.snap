// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoComponent renders the text inside it 1`] = `ReactWrapper {}`;
